### PR TITLE
Fix: CloudConnection did not forward close() call to cloud proxy

### PR DESCRIPTION
**Pull request by vipeller**
The main change is in BrokeredCloudConnection.cs where it did not forward the call to CloudProxy. The rest of the change is only about to being able to log the Identity in case of failure.
The removed "TODO" about the subscription is not needed anymore as the proper place was in ConnectionHander.cs/RemoveConnectionAsync() to remove existing subscriptions.

### CODE REVIEW

File: edge-hub/core/src/Microsoft.Azure.Devices.Edge.Hub.MqttBrokerAdapter/upstream/BrokeredCloudProxy.cs [hunk 1]
```diff
@@ -14,22 +14,22 @@ namespace Microsoft.Azure.Devices.Edge.Hub.MqttBrokerAdapter
...
-        IIdentity identity;
         Action<string, CloudConnectionStatus> connectionStatusChangedHandler;
 
         AtomicBoolean isActive = new AtomicBoolean(true);
         AtomicBoolean twinNeedsSubscribe = new AtomicBoolean(true);
 
         public BrokeredCloudProxy(IIdentity identity, BrokeredCloudProxyDispatcher cloudProxyDispatcher, Action<string, CloudConnectionStatus> connectionStatusChangedHandler)
         {
-            this.identity = Preconditions.CheckNotNull(identity);
+            this.Identity = Preconditions.CheckNotNull(identity);
```
Comment (vadim-kovalyov) [inline edge-hub/core/src/Microsoft.Azure.Devices.Edge.Hub.MqttBrokerAdapter/upstream/BrokeredCloudProxy.cs:24]: nit: why change this to prop?

**Review by vadim-kovalyov — APPROVED** [review]
LGTM

File: edge-hub/core/src/Microsoft.Azure.Devices.Edge.Hub.MqttBrokerAdapter/upstream/BrokeredCloudConnection.cs [hunk 2]
```diff
@@ -4,26 +4,58 @@ namespace Microsoft.Azure.Devices.Edge.Hub.MqttBrokerAdapter
...
         {
-            // TODO, in order to tear down the connection higher level,
-            // it should unsubscribe from all topics
             this.IsActive = false;
-            return Task.FromResult(true);
+            var result = default(bool);
+
+            try
+            {
+                result = await this.CloudProxy.Match(
```
Comment (varunpuranik) [inline edge-hub/core/src/Microsoft.Azure.Devices.Edge.Hub.MqttBrokerAdapter/upstream/BrokeredCloudConnection.cs:35]: Nit: suggest: .Map().GetOrElse(true)

Comment (varunpuranik) [inline edge-hub/core/src/Microsoft.Azure.Devices.Edge.Hub.MqttBrokerAdapter/upstream/BrokeredCloudConnection.cs:35]: test?

File: edge-hub/core/src/Microsoft.Azure.Devices.Edge.Hub.MqttBrokerAdapter/upstream/BrokeredCloudConnection.cs [hunk 3]
```diff
@@ -4,26 +4,58 @@ namespace Microsoft.Azure.Devices.Edge.Hub.MqttBrokerAdapter
...
     using Microsoft.Azure.Devices.Edge.Hub.Core.Cloud;
+    using Microsoft.Azure.Devices.Edge.Hub.Core.Identity;
     using Microsoft.Azure.Devices.Edge.Util;
+    using Microsoft.Extensions.Logging;
 
     public class BrokeredCloudConnection : ICloudConnection
     {
+        IIdentity identity;
+
         public BrokeredCloudConnection(BrokeredCloudProxy cloudProxy)
```
Comment (varunpuranik) [inline edge-hub/core/src/Microsoft.Azure.Devices.Edge.Hub.MqttBrokerAdapter/upstream/BrokeredCloudConnection.cs:15]: Why is this not a ICloudProxy? We should avoid passing in actual types.. if possible.

File: edge-hub/core/src/Microsoft.Azure.Devices.Edge.Hub.MqttBrokerAdapter/upstream/BrokeredCloudConnection.cs [hunk 4]
```diff
@@ -4,26 +4,58 @@ namespace Microsoft.Azure.Devices.Edge.Hub.MqttBrokerAdapter
...
         {
+            Preconditions.CheckNotNull(cloudProxy);
+
             this.IsActive = true;
             this.CloudProxy = Option.Some(cloudProxy as ICloudProxy);
+            this.identity = cloudProxy.Identity;
```
Comment (varunpuranik) [inline edge-hub/core/src/Microsoft.Azure.Devices.Edge.Hub.MqttBrokerAdapter/upstream/BrokeredCloudConnection.cs:21]: This seems odd... can we just pass in the Identity separately, like we do for CloudConnection?

**Review by nyanzebra — COMMENTED** [review]
Let's put at least 1 UT covering the change, if we need more thorough tests, we can add those in another pr if that is easier or not a high priority. Will approve after that :)

**Review by nyanzebra — APPROVED** [review]
LGTM 🚢

**Pull request merged**